src/server/models/order.ts

```diff
@@ -42,20 +47,50 @@ class Order extends Model<Order> {
       .then(async orders => {
         let cart
         if (orders.length === 0) {
-          console.log('cannot find order: creating one')
+          console.log(`cannot find order for userId ${userId}: creating one`)
           cart = await Order.create({ userId, status: Status.Cart })
         } else {
           cart = orders.find(order => order.status === Status.Cart)
           if (!cart) {
-            console.log('cannot find cart: creating one')
+            console.log(`cannot find cart for userId : creating one`)
             cart = await Order.create({ userId, status: Status.Cart })
           }
         }
         return cart
       })
-      .then(cart => OrdersProducts.create({ orderId: cart.id, productId: productId }))
+      // 4. find existing ordersProduct instance and increase quantity by quantity.
+      // 5. if no existing ordersProduct instance, create instance.
+      .then(cart => OrdersProducts.create({ orderId: cart.id, productId: productId, quantity: quantity }))
```

Comment:
Does this check if the order - product combination exists, and (if so), increases the quantity?